contracts/HubPool.sol

```diff
@@ -375,39 +382,114 @@ contract HubPool is HubPoolInterface, Testable, Lockable, MultiCaller, Ownable {
+     *     - whitelist deposit route from USDC on Optimism to USDC on Arbitrum
+     *     - whitelist deposit route from USDC on Arbitrum to USDC on Optimism
+     * @param depositRouteChainId_1 Chain ID for one SpokePool that we want to enable as a deposit origin and
+     * destination. Pool rebalances to this chain will also be enabled.
+     * @param depositRouteChainId_2 The other chain in addition to `depositRouteChainId_1` that we want to enable
+     * as a deposit origin and destination, and make available for pool rebalances.
+     * @param ethereumCounterpartToken Token on the current network that will be sent to the SpokePool on
+     * both `depositRouteChainId_1` and ``depositRouteChainId_2` for pool rebalances.
+     * @param l2Token_1 Token that we want to enable for deposits on `depositRouteChainId_1` that can be fulfilled by
+     * `l2Token_2` on `depositRouteChainId_2`. Should be the chainId_1 counterpart of `ethereumCounterpartToken`.
+     * @param l2Token_2 Token that we want to enable for deposits on `depositRouteChainId_2` that can be fulfilled by
+     * `l2Token_1` on `depositRouteChainId_1`. Should be the chainId_2 counterpart of `ethereumCounterpartToken`.
+     * @param enable Set to True to set up pool rebalances to `depositRouteChainId_1` and `depositRouteChainId_2` and
+     * deposits from `depositRouteChainId_1` to `depositRouteChainId_2`. Set to False to disable all of the above. If
+     * this value is false, the pool rebalance destination tokens will be set to 0x0 and the deposit routes will be
+     * disabled on the SpokePools.
+     */
+    function setDepositAndPoolRebalanceRoute(
+        uint256 depositRouteChainId_1,
+        uint256 depositRouteChainId_2,
+        address ethereumCounterpartToken,
+        address l2Token_1,
+        address l2Token_2,
+        bool enable
+    ) public override nonReentrant onlyOwner {
+        poolRebalanceRoutes[_poolRebalanceRouteKey(ethereumCounterpartToken, depositRouteChainId_1)] = (
+            enable ? l2Token_1 : address(0)
+        );
+        poolRebalanceRoutes[_poolRebalanceRouteKey(ethereumCounterpartToken, depositRouteChainId_2)] = (
+            enable ? l2Token_2 : address(0)
+        );
+        _relaySpokePoolAdminFunction(
+            depositRouteChainId_1,
+            abi.encodeWithSignature(
+                "setEnableRoute(address,address,uint256,bool)",
+                l2Token_1,
+                l2Token_2,
+                depositRouteChainId_2,
+                enable
+            )
+        );
+        _relaySpokePoolAdminFunction(
+            depositRouteChainId_2,
+            abi.encodeWithSignature(
+                "setEnableRoute(address,address,uint256,bool)",
+                l2Token_2,
+                l2Token_1,
+                depositRouteChainId_1,
+                enable
+            )
+        );
+
+        emit SetPoolRebalanceRoute(depositRouteChainId_1, ethereumCounterpartToken, l2Token_1);
```

Comment:
If you are going to whitelist/enable routes in both directions would it make sence to create some kind of condensed event rather than emitting multiple times?

Reply:
I think clients should be able to reliably query `SetEnableDepositRoute` and `SetPoolRebalanceRoute` without caring how they were emitted (i.e. via a batch method or not)

Reply:
I did end up putting these emits into internal functions so you won't see multiple emits in a single method anymore